### DISCUSSION TITLE
JBrowse 2 React App Demo：TypeError: Cannot read properties of undefined (reading 'length')

### DISCUSSION THREAD

**Discussion by MorganHis** [GMOD/jbrowse-components#3987] [Q&A] [answered]
When I use the Open session function of the JBrowse 2 React App Demo menu bar, an error occurs: TypeError: Cannot read properties of undefined (reading 'length')
<img width="2555" alt="image" src="https://github.com/GMOD/jbrowse-components/assets/84215074/c7c29975-2e08-41f5-a253-fdc153d5e849">
<img width="2558" alt="image" src="https://github.com/GMOD/jbrowse-components/assets/84215074/20917fdb-6b17-4c54-b580-bb40fddd83fd">

**Comment by MorganHis** [comment]
In addition, when I used the Duplicate session function in the JBrowse 2 React App Demo menu bar, another error occurred: Uncaught TypeError: e.root.duplicateCurrentSession is not a function at Object.duplicateCurrentSession.
<img width="2557" alt="image" src="https://github.com/GMOD/jbrowse-components/assets/84215074/3113744a-42ab-414a-9837-8b4d03bfe5f4">

**Reply by cmdcolin** [reply, marked as answer]
thanks for reporting this, it's indeed a bug! will make a fix. it's possible we could remove these functions for the app, or find a replacement (the savedSessions concept assumes use of localstorage in the jbrowse-web app but for @jbrowse/react-app, we try not to assume too many things about how sessions are 'saved')

**Reply by MorganHis** [reply]
It would be nice to remove these functions, thanks for the reply!

**Reply by cmdcolin** [reply]
I just released v2.7.1. It removed the functions that were broken...hopefully that helps for now, but let me know if there is any futher issue!

**Reply by MorganHis** [reply]
Thank you very much, I have used v2.7.1, and I am happy to see that the "Duplicate session" function and "Open session" function have been deleted, but I found errors in the other two functions of the menu bar, you can check details in this new discussion (https://github.com/GMOD/jbrowse-components/discussions/4017#discussion-5774929). Thank you for your efforts!
![image](https://github.com/GMOD/jbrowse-components/assets/84215074/0f8eabbf-a030-4cce-89c2-1f047b97a24d)

**Comment by MorganHis** [comment]
[https://jbrowse.org/demos/app-cra5/](url)⬅️Link to JBrowse 2 React App Demo
